project/src/main/webapp/script.js

```diff
@@ -12,47 +12,106 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.import java.io.IOException;
 
+/* The user location map. the map has to be accessed from different functions, so it has to be kept
+ * globally. */
+let globalUserMap;
+
 /**
  * Fetches recommended places from the 'query' servlet, and switches from the query form to the
  * results elements in order to display them to the user.
  */
 function fetchFromQuery() {
-  document.getElementById('query-form').style.display = 'none';
-  document.getElementById('results').style.display = 'block';
-  const placesDiv = document.getElementById('place');
-  fetch('/query').then(response => response.json()).then((places) => {
-    places.forEach((singlePlace) => {
-      placesDiv.appendChild(createPlaceElement(singlePlace));
+  try {
+    const params = [
+      `quisines=${getUserQuisinesFromUi()}`,
+      `rating=${getUserRatingFromUi()}`,
+      `price=${getUserPriceFromUi()}`,
+      `open=${getUserOpenNowFromUi()}`,
+      `location=${getUserLocationFromUi()}`
+    ].join('&');
+    const placesDiv = document.getElementById('place');
+    fetch('/query?' + params).then(response => response.json()).then((places) => {
+      places.forEach((singlePlace) => {
+        placesDiv.appendChild(createPlaceElement(singlePlace));
+      });
     });
-  });
+    document.getElementById('query-form').style.display = 'none';
+    document.getElementById('results').style.display = 'block';
+  } catch (error) {
+    alert(error.message);
+  }
+}
+
+function getUserQuisinesFromUi() {
+  const quisines = document.getElementById('quisines-form').elements;
+  let result = '';
+  let i;
+  for (i = 0; i < quisines.length; i++) {
+    if (quisines[i].checked) {
+      result = result + quisines[i].value + ',';
+    }
+  }
+  if (result === '') {
+    throw new Error("Choose at least one quisine.");
```

Comment:
cuisine. It also need to be fixed in other places

Reply:
Hahaha oh my god. Definitely